tracer/src/Datadog.Trace/Scope.cs

```diff
@@ -13,23 +13,29 @@ namespace Datadog.Trace
     /// all newly created spans that are not created with the ignoreActiveSpan
     /// parameter will be automatically children of the active span.
     /// </summary>
-    public class Scope : IDisposable
+    public class Scope : IScope
     {
         private readonly IScopeManager _scopeManager;
         private readonly bool _finishOnClose;
 
-        internal Scope(Scope parent, Span span, IScopeManager scopeManager, bool finishOnClose)
+        internal Scope(Scope parent, ISpan span, IScopeManager scopeManager, bool finishOnClose)
         {
             Parent = parent;
             Span = span;
+            InternalSpan = span as Span;
```

Comment:
In mixed version scenarios, this could be null right, as it could be a proxy instance? I assume it can, as `Tracer.ActivateSpan(ISpan)` is a public API which ultimately calls this. In which case, this will be null if it's invoked from that API.

I'm not sure if there's a way round it, but it makes me nervous that we _know_ `InternalSpan` can be null, yet in most places (everywhere?) in this PR we blindly access it assuming it won't be😟

Reply:
In the latest set of commits, I removed the member `Scope.InternalSpan` so the change is only that `Scope.Span` has type `ISpan`. This should remove concerns about nullability. Now at runtime we still don't know what runtime type `Scope.Span` actually is, so we have to test against `Scope.Span is Span` returning a null or non-null value